### PR TITLE
use miniflare/workerd

### PR DESCRIPTION
**Pull request by threepointone**
This changes the local dev experience to use cloudflare's js runtime workerd (via miniflare). This has a lot of implications:
- the dev runtime now matches closer to what's running in production
- this means we'll get platform features more quickly (as an example, we can now connectto postgres databases)
- we can iterate on our features more quickly, since we don't have to maintain 2 separate implementations
- fewer bugs between dev and prod

The tradeoffs -
- it's quite heavy, so initial install is going to be slow
- won't run in stackblitz-like environments
- it's a brand new runtime, so we can expect some bugs
- setting up types is kind of annoying
- setting up a test environment is annoying

---

TODO: 
- [x] fix the types story, I have way too many expect-errors floating around 
- [x] when running the lexical example, this shows up: `An event handler returned a promise that will be ignored. Event handlers should not have a return value and should not be async functions.`
- [ ] this warning when running eslint `WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.`

### CODE REVIEW

File: examples/basic/src/client.ts [hunk 1]
```diff
@@ -26,6 +27,7 @@ setInterval(() => {
   partySocket.send(`latency:${id}`);
 }, 1000);
 
+// @ts-expect-error I dunno how to type this
```
Comment (sadn1ck) [inline examples/basic/src/client.ts:30]: adding "DOM" to examples/basic/tsconfig.json > compilerOptions.lib array should fix this (and the other ones as well. [base templates from vite](https://github.com/vitejs/vite/blob/471fba2d1cf647a0bea747739d4d90def60e83ce/packages/create-vite/template-react-ts/tsconfig.json#L5) have it, for reference)

docs: https://www.typescriptlang.org/tsconfig#lib

File: examples/blocknote/src/client.tsx [hunk 2]
```diff
@@ -34,6 +34,7 @@ function Editor() {
   return <BlockNoteView editor={editor} />;
 }
 
+// @ts-expect-error I dunno how to fix this type TODO fix this
 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 const root = createRoot(document.getElementById("root")!);
```
Comment (sadn1ck) [inline examples/blocknote/src/client.tsx:39]: same "DOM" to compilerOptions.lib fix + assertion as `HTMLElement` ([ref to vite template](https://github.com/vitejs/vite/blob/471fba2d1cf647a0bea747739d4d90def60e83ce/packages/create-vite/template-react-ts/src/main.tsx#L6))

File: examples/basic/src/client.ts (hunk 1 above)
Comment (threepointone) [reply examples/basic/src/client.ts:30]: I tried this, unfortunately it clashes with some types in workers-types (specifically, 'globals' like `Request`/`Response` that have slight differences in the server environment. Ideally I'd like to be able to specify different types for `client.ts` and `server.ts` without making them separate packages. Any ideas?

Comment (sadn1ck) [reply examples/basic/src/client.ts:30]: I think we can just have separate `tsconfig` for the client and server each inside a folder without splitting it into different packages (by split I assume you meant basic-client + basic-server)
image

although can't speak for the maintainability further down the line haha

Comment (threepointone) [reply examples/basic/src/client.ts:30]: I wouldn't mind doing this, but now I can't just call `tsc` in the root here to do a full typecheck (or can I?)

Comment (sadn1ck) [reply examples/basic/src/client.ts:30]: hmm yup you're correct, we can't. that's a problem

**Comment by threepointone** [comment]
Landing this after a quick swim, lfg.

**Pull request merged**